catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/helper/RegistryPackageTypeHelper.java

```diff
@@ -57,241 +68,317 @@ public void setRegistryPackage(RegistryPackageType registryPackage) {
+
+        return getObjectsFromRegistryObjectList(registryPackage, type).stream()
+                .filter(isAssociatedObject(targetObjectIds))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * This is a convenience method that returns all Objects associated with the provided id extracted from the provided RegistryObjectList
+     *
+     * @param registryObjectList the RegistryObjectList that will be crawled, null returns an empty list
+     * @param id                 to be used as the sourceObjectId of the Association
+     * @param type               Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(
+            RegistryObjectListType registryObjectList, String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
+            return Collections.emptyList();
+        }
+
+        Set<String> targetObjectIds = getAssociationTargetObjectIds(registryObjectList, id);
+        if (CollectionUtils.isEmpty(targetObjectIds)) {
+            return Collections.emptyList();
+        }
+
+        return getObjectsFromRegistryObjectList(registryObjectList, type).stream()
+                .filter(isAssociatedObject(targetObjectIds))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * This is a convenience method that returns all Objects associated with the provided id extracted from this class's RegistryPackageType
+     *
+     * @param id   to be used as the sourceObjectId of the Association
+     * @param type Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
+            return Collections.emptyList();
+        }
+        Set<String> targetObjectIds = getAssociationTargetObjectIds(id);
+        if (CollectionUtils.isEmpty(targetObjectIds)) {
+            return Collections.emptyList();
+        }
+
+        return getObjectsFromRegistryObjectList(registryPackageType, type).stream()
+                .filter(isAssociatedObject(targetObjectIds))
+                .collect(Collectors.toList());
+    }
+
+    private Predicate<RegistryObjectType> isAssociatedObject(Set<String> associatedIds) {
+        return p -> associatedIds.contains(p.getId());
+    }
+
+    private Set<String> getAssociationTargetObjectIds(RegistryPackageType registryPackage,
+            String sourceId) {
+        Set<String> targetObjectIds = new HashSet<>();
+
+        if (registryPackage == null) {
+            return targetObjectIds;
+        }
+
+        if (registryPackage.isSetRegistryObjectList()) {
+            targetObjectIds = getAssociationTargetObjectIds(registryPackage.getRegistryObjectList(), sourceId);
+        }
+        return targetObjectIds;
```

Comment:
I think you can avoid declaring/instantiating the Set by combining the conditional checks and returning the results of `getAssociationTargetObjectIds`, `new HashSet<>()` otherwise. It is clear what is happening as is though.